tests/ui/type-alias-impl-trait/issue-52843-closure-constrain.rs

```diff
@@ -1,13 +1,15 @@
 // Checks to ensure that we properly detect when a closure constrains an opaque type
 
-#![feature(type_alias_impl_trait)]
+#![feature(type_alias_impl_trait, stmt_expr_attributes)]
 
 use std::fmt::Debug;
 
 fn main() {
     type Opaque = impl Debug;
+    #[define_opaques(Opaque)]
     fn _unused() -> Opaque { String::new() }
-    let null = || -> Opaque { 0 };
+    let null = #[define_opaques(Opaque)] || -> Opaque { 0 };
```

Comment:
do u need that attr? I would have expected `main` to already be a defining scope of `Opaque` as it contains the TAIT?

Reply:
This attribute is a noop on closures and just produces an error.

Similarly, as discussed in the types team meeting, we decided to just not allow this for now. You should move your TAIT out of the function and then mark the function as `define_opaque` for that TAIT.

Reply:
> This attribute is a noop on closures and just produces an error.

Then why add it to the closure :thinking: to have a regression test if we ever want to support defining opaques exclusively in a nested item, but not the parent?

I consider that unlikely and think we'll definitely come up with such a test again if we ever do. I would like to rm that test